src/linquantarrays.jl

```diff
@@ -10,84 +13,164 @@ Base.size(QA::LinQuantArray) = size(QA.A)
 Base.Array{T}(Q::LinQuantArray{UInt24,N}) where {T,N} = Array{T}(24,Q)
 Base.Array{T}(Q::LinQuantArray{UInt32,N}) where {T,N} = Array{T}(32,Q)
 
-Base.Array(Q::LinQuantArray{UInt8,N}) where N = Array{Float32}(8,Q)
-Base.Array(Q::LinQuantArray{UInt16,N}) where N = Array{Float32}(16,Q)
-Base.Array(Q::LinQuantArray{UInt24,N}) where N = Array{Float32}(24,Q)
-Base.Array(Q::LinQuantArray{UInt32,N}) where N = Array{Float64}(32,Q)
+# define default conversions for signed 8, 16, 24 and 32 bit
+Base.Array(Q::LinQuantArray{Int8,N}) where N = Array{Float32}(8,Q)
+Base.Array(Q::LinQuantArray{Int16,N}) where N = Array{Float32}(16,Q)
+Base.Array(Q::LinQuantArray{Int24,N}) where N = Array{Float32}(24,Q)
+Base.Array(Q::LinQuantArray{Int32,N}) where N = Array{Float64}(32,Q)
 
 # one quantization per layer
 """Linear quantization independently for every element along dimension
 dim in array A. Returns a Vector{LinQuantArray}."""
-function LinQuantArray(::Type{TUInt},A::AbstractArray{T,N},dim::Int) where {TUInt,T,N}
+function LinQuantArray(
+    ::Type{TInteger},
+    A::AbstractArray{T,N},
+    dim::Int,
+    extrema::Option{Tuple} = nothing
+) where {TInteger,T,N}
     @assert dim <= N   "Can't quantize a $N-dimensional array in dim=$dim"
     n = size(A)[dim]
     L = Vector{LinQuantArray}(undef,n)
     t = [if j == dim 1 else Colon() end for j in 1:N]
     for i in 1:n
         t[dim] = i
-        L[i] = LinQuantization(TUInt,A[t...])    
+        L[i] = LinQuantization(TInteger,A[t...], extrema)    
     end
     return L
 end
 
-# for 8,16,24 and 32 bit
+# for unsigned integers  8,16,24 and 32 bit
 LinQuant8Array(A::AbstractArray{T,N},dim::Int) where {T,N} = LinQuantArray(UInt8,A,dim)
 LinQuant16Array(A::AbstractArray{T,N},dim::Int) where {T,N} = LinQuantArray(UInt16,A,dim)
 LinQuant24Array(A::AbstractArray{T,N},dim::Int) where {T,N} = LinQuantArray(UInt24,A,dim)
 LinQuant32Array(A::AbstractArray{T,N},dim::Int) where {T,N} = LinQuantArray(UInt32,A,dim)
 
+# for unsigned integers  8,16,24 and 32 bit
+LinQuantUInt8Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(UInt8,A,dim,e)
+LinQuantUInt16Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(UInt16,A,dim,e)
+LinQuantUInt24Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(UInt24,A,dim,e)
+LinQuantUInt32Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(UInt32,A,dim,e)
+
+# for signed integers of 8,16,24 and 32 bit
+LinQuantInt8Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(Int8,A,dim,e)
+LinQuantInt16Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(Int16,A,dim,e)
+LinQuantInt24Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(Int24,A,dim,e)
+LinQuantInt32Array(A::AbstractArray{T,N},dim::Int,e::Option{Tuple}) where {T,N} = LinQuantArray(Int32,A,dim,e)
```

Comment:
We should be able to define these simply with the parameter `T` that `LinQuantArray{T}` already has. So like use `LinQuant8Array` for unsigned 8-bit integers, but if you want signed use `LinQuantArray{Int8}` that makes the unsigned integers appear as default but it's still convenient to write for signed integers without defining 8 new function names that are all exported and that basically just double up the type definition